### PR TITLE
Text block comments

### PR DESCRIPTION
**Pull request by achyu-dev**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,7 +59,7 @@
     "ts-node": "10.9.2",
     "tsx": "^4.19.2",
     "typescript": "^5.7.3",
-    "vitest": "1.6.0"
+    "vitest": "^1.6.1"
```
Comment (vcarl) [inline package.json:62]: Why is this changing?

File: src/features/emojiMod.ts [hunk 2]
```diff
@@ -44,6 +43,23 @@ type ReactionHandlers = {
   }) => void;
 };
 
+async function waitForThreadActivity(
```
Comment (vcarl) [inline src/features/emojiMod.ts:46]: What's this change? Seems unrelated from the main intent of this PR, can this be pulled out into a separate change with its own description?

File: src/features/emojiMod.ts [hunk 3]
```diff
@@ -110,21 +139,24 @@ Zell Liew [wrote a great resource](https://zellwk.com/blog/asking-questions/) ab
...
+      await thread.send(truncateMessage(message.content));
+
+      await message.delete();
+      reportUser({
+        reason: ReportReasons.lowEffortQuestionRemoved,
+        message,
+        staff: staffOrHelpfulReactors,
+      });
+    } catch (error) {
+      console.error(`Failed to create private thread: ${error}`);
```
Comment (vcarl) [inline src/features/emojiMod.ts:158]: Can this be try/catch undone, please? Touching unrelated code, even for well-intention changes like improving logging, increases the scope of what must be tested and increases the risk of introducing bugs

File: src/features/emojiMod.ts [hunk 4]
```diff
@@ -147,30 +178,39 @@ const emojiMod: ChannelHandlers = {
...
-      guild,
-      author: authorMember,
-      reactor,
-      message: fullMessage,
-      reaction: fullReaction,
-      usersWhoReacted: usersWhoReacted.filter(
-        (x): x is GuildMember => Boolean(x) && authorMember.id !== reactor.id,
-      ),
-    });
+    try {
```
Comment (vcarl) [inline src/features/emojiMod.ts:181]: likewise

File: test/commands.test.ts [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+  it("should ignore commands inside triple backticks", () => {
+    mockMessage.content = "```\n!mdn Array.prototype.map\n```";
+    expect(shouldProcessCommand(mockMessage.content, "!mdn")).toBe(false);
+  });
+
+  it("should ignore commands inside nested backticks", () => {
+    mockMessage.content =
+      "```js\nconsole.log(`!mdn Array.prototype.map`);\n```";
+    expect(shouldProcessCommand(mockMessage.content, "!mdn")).toBe(false);
+  });
```
Comment (vcarl) [inline test/commands.test.ts:92]: Please add some more complex messages to test here — you're only testing the "happy path" here, not the edge cases

File: {
Comment (vcarl) [inline {:1]: Remove please

**Review by vcarl — CHANGES_REQUESTED** [review]
Hey, a couple of required changes before I can approve this — let's keep the PR narrowly focused on just the specific issue of command triggers in code blocks

File: src/features/emojiMod.ts (hunk 3 above)
Comment (achyu-dev) [reply src/features/emojiMod.ts:158]: yah, I retained back the original `emojiMod.ts` file, I really didnt modify this file

File: {
Comment (achyu-dev) [reply {:1]: remove what?

File: package-lock.json
Comment (vcarl) [inline package-lock.json:1]: 👀 this still reflects the vitest version bump that was removed

File: {
Comment (vcarl) [reply {:1]: The empty file named `{`

File: src/features/commands.ts [hunk 6]
```diff
@@ -1257,6 +1257,42 @@ const createCommandsMessage = () => {
   return categoryDescriptions.join("\n\n").trim();
 };
 
+export function removeCodeBlocksAndQuotes(text: string): string {
```
Comment (vcarl) [inline src/features/commands.ts:1260]: Is this function used? 🤔 I don't see it referenced elsewhere in the PR

File: src/features/commands.ts [hunk 7]
```diff
@@ -1257,6 +1257,42 @@ const createCommandsMessage = () => {
...
+    if (text.slice(i, i + 3) === "```") {
+      insideTriple = !insideTriple;
+      i += 3;
+    } else if (text[i] === "`" && !insideTriple) {
+      insideSingle = !insideSingle;
+      i++;
+    } else {
+      i++;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (vcarl) [reply src/features/commands.ts:1285]: Actually it's more resource intensive than I thought — it will create n substrings for every command trigger, and we have ~ 40 commands with ~ 50 triggers. So that means that a message with 150 characters will create 150 3-character substrings 50 times — I think this is a no-go the way it's written

[the remark this answers is not in the document]
Comment (achyu-dev) [reply src/features/commands.ts:1285]: Hmm okay, I will think of a better approach for this

File: {
[the remark this answers is not in the document]
Comment (achyu-dev) [reply {:1]: let me check

[the remark this answers is not in the document]
Comment (achyu-dev) [reply {:1]: Ah yes, it was created accidentally, I deleted now

File: src/features/commands.ts (hunk 6 above)
Comment (achyu-dev) [reply src/features/commands.ts:1260]: yep it was not used, u can see it being used used in the next commit

File: package-lock.json
Comment (achyu-dev) [reply package-lock.json:1]: Didnt get you. Can u quote the changes?

Comment (vcarl) [reply package-lock.json:1]: `package-lock.json` should only change if there are changes to dependencies. Since the version changes in `package.json` got reverted, there shouldn't be any modifications to `package-lock.json` in this PR. You'll probably need to run `npm i` again to update it

File: src/features/commands.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (kristersd) [reply src/features/commands.ts]: I wonder, couldn't you use `replaceAll` to remove all content in-between backticks ` (other forbidden backticks). Then you could compare, if the trigger name is still there.

[the remark this answers is not in the document]
Comment (achyu-dev) [reply src/features/commands.ts]: shud be possible, will try it @kristersd . thanks !!

[the remark this answers is not in the document]
Comment (achyu-dev) [reply src/features/commands.ts]: In the meantime, check the latest commit @kristersd if tats an optimal approach

**Comment by DanielFGray** [comment]
Seems like this could be made a lot simpler utilizing a slightly more complex regex: 
```
/!\w+(?=[^`]*(?:`[^`]*`[^`]*)*$)/
```
this should match `!words` not inside code blocks
in action: https://regex101.com/r/pm4kqS/1

**Comment by achyu-dev** [comment]
Can someone help me on why these 8 test cases fail?

```
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 8 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  test/commands.test.ts > Discord Bot Commands > should ignore command inside deeply nested triple backticks
AssertionError: expected true to be false // Object.is equality

- Expected
+ Received

- false
+ true

 ❯ test/commands.test.ts:129:63
    127|       "```\nSome random text\n```js\nconsole.log('!mdn Array.prototype.map');\n```\nMore text outside";
    128|
    129|     expect(shouldProcessCommand(mockMessage.content, "!mdn")).toBe(false);
       |                                                               ^
    130|   });
    131|

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/8]⎯

 FAIL  test/commands.test.ts > Discord Bot Commands > should ignore command inside a long deeply nested code block
AssertionError: expected true to be false // Object.is equality

- Expected
+ Received

- false
+ true

 ❯ test/commands.test.ts:145:63


⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/8]⎯

 FAIL  test/commands.test.ts > Discord Bot Commands > should process command outside deeply nested backticks but ignore the inside one
AssertionError: expected false to be true // Object.is equality

- Expected
+ Received

- true
+ false

 ❯ test/commands.test.ts:177:63
    175|       "Hey everyone, I need help with JavaScript! ```js\nfunction test() {\n  console.log('!mdn Array.prototype.map');\n}\n``` But I still need info on !m…
    176|
    177|     expect(shouldProcessCommand(mockMessage.content, "!mdn")).toBe(true); // The one outside should be processed.
       |                                                               ^
    178|     expect(
    179|       shouldProcessCommand(

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/8]⎯

 FAIL  test/commands.test.ts > Discord Bot Commands > should ignore command inside multi-level text and markdown blocks
AssertionError: expected true to be false // Object.is equality

- Expected
+ Received

- false
+ true

 ❯ test/commands.test.ts:193:63
    191|       "```\nAnother block\n```js\nconsole.log('!mdn Array.prototype.map');\n```";
    192|
    193|     expect(shouldProcessCommand(mockMessage.content, "!mdn")).toBe(false);
       |                                                               ^
    194|   });
    195|

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[4/8]⎯

 FAIL  test/commands.test.ts > Discord Bot Commands > should ignore command if inside an embedded link
AssertionError: expected true to be false // Object.is equality

- Expected
+ Received

- false
+ true

 ❯ test/commands.test.ts:247:63
    245|     mockMessage.content =
    246|       "[Click here for !mdn Array.prototype.map](https://example.com)";
    247|     expect(shouldProcessCommand(mockMessage.content, "!mdn")).toBe(false);
       |                                                               ^
    248|   });
    249|

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[5/8]⎯

 FAIL  test/commands.test.ts > Discord Bot Commands > should ignore command inside nested code blocks
AssertionError: expected true to be false // Object.is equality

- Expected
+ Received

- false
+ true

 ❯ test/commands.test.ts:261:63
    259|     mockMessage.content =
    260|       "```\nHere is some code:\n```js\nconsole.log('!mdn Array.prototype.map');\n```";
    261|     expect(shouldProcessCommand(mockMessage.content, "!mdn")).toBe(false);
       |                                                               ^
    262|   });
    263|

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[6/8]⎯

 FAIL  test/commands.test.ts > Discord Bot Commands > should ignore command if it's part of a quoted message
AssertionError: expected true to be false // Object.is equality

- Expected
+ Received

- false
+ true

 ❯ test/commands.test.ts:273:63
    271|   it("should ignore command if it's part of a quoted message", () => {
    272|     mockMessage.content = `> Someone said: "!mdn Array.prototype.map is useful"`;
    273|     expect(shouldProcessCommand(mockMessage.content, "!mdn")).toBe(false);
       |                                                               ^
    274|   });
    275|

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[7/8]⎯

 FAIL  test/commands.test.ts > Discord Bot Commands > should ignore command if part is inside a code block
AssertionError: expected false to be true // Object.is equality

- Expected
+ Received

- true
+ false

 ❯ test/commands.test.ts:323:63
    321|     mockMessage.content =
    322|       "```js\nconsole.log('!mdn Array.prototype.map');\n``` !mdn Array.prototype.map";
    323|     expect(shouldProcessCommand(mockMessage.content, "!mdn")).toBe(true); // Only process the second one
       |                                                               ^
    324|   });
    325| });

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[8/8]⎯

 Test Files  1 failed | 5 passed (6)
      Tests  8 failed | 48 passed | 1 skipped (57)
   Start at  12:21:41
   Duration  1.40s (transform 378ms, setup 1ms, collect 1.57s, tests 504ms, environment 1ms, prepare 936ms)
```
